prboom2/src/dsda/dgeom.h

```diff
@@ -0,0 +1,212 @@
+//
+// Copyright(C) 2023 Brian Koropoff
+//
+// This program is free software; you can redistribute it and/or
+// modify it under the terms of the GNU General Public License
+// as published by the Free Software Foundation; either version 2
+// of the License, or (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+// DESCRIPTION:
+//	Internal double-precision geometry functions
+//
+//      Not intended for use outside of DSDA subdir, so these aren't
+//      prefixed with dsda_
+//
+
+#ifndef DSDA_INTERNAL_
+#error Internal header, not for use outside dsda subdir
+#endif
+
+#ifndef DSDA_DGEOM_H_
+#define DSDA_DGEOM_H_
+
+#include <math.h>
+
+#include "m_fixed.h"
+#include "r_defs.h"
+#include "r_state.h"
+
+struct dpoint
```

Comment:
Please typedef all structs. I.e., set this to a `dpoint_t` and reference it as such when possible.